src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp

```diff
@@ -184,6 +149,11 @@ void QgsValueRelationWidgetWrapper::setValue( const QVariant &value )
   {
     QStringList checkList( QgsValueRelationFieldFormatter::valueToStringList( value ) );
 
+    QTableWidgetItem *lastChangedItem = nullptr;
+
+    // This block is needed because item->setCheckState triggers dataChanged gets back to value()
+    // and iterate over all items again! This can be extremely slow on large items sets.
+    mTableWidget->blockSignals( true );
```

Comment:
Would it be possible to use a variable to guard from this? I'm always afraid of unintended side-effects of blocking all signals of an object.

Reply:
I'm not sure I get what you mean, the logic here is to block signals during the loop and fire the signal only once for the last item.
My test case was probably a worst case scenario where the referenced layer had hundreds of items, this means that in case of AllowMulti we had to loop through hundred of items and set the checkbox for each of them and fire the signal (which is not just one but a chain of signals) for each of them, this proved to be extremely slow, that's why I decided to block the signals in the loop and fire just once for the last item, this solved the performance issue but retains the original behavior because the signal is still fired (just once instead of hundred of times).

Reply:
The risk with this approach is that not only the problematic signal is blocked (which you take care of emitting in the end) but every signal. I don't know which other ones it would be here at the moment and no one knows which ones are introduced in the future. That's why I think it's often a good idea to set a member variable (`bool mIsUpdatingWidget`, better name welcome) and check in the slot if it is set and exit. But this only works if it's one of our slots and not a Qt slot.

Reply:
No, I guess this falls into the category where it doesn't work.